frontend/src/app/components/ExperimentPage.js

```diff
@@ -127,6 +129,10 @@ export default class ExperimentPage extends React.Component {
       statusType = 'enabled';
     }
 
+    if (moment(utcNow).isBefore(experiment.launch_date) && typeof experiment.launch_date !== 'undefined') {
+      this.props.history.push('/not-found');
```

Comment:
Don't use `this.props.history.push` directly, here. Instead use the `navigateTo` store dispatch helper. You probably also want to return empty content, rather then carry on to render the rest of the page. Maybe, something like:
```
navigateTo('/not-found');
return '';
```